pkg/yurthub/configuration/manager.go

```diff
@@ -0,0 +1,289 @@
+						}
+					}
+				}
+			}
+		}
+	}
+	// init filter settings
+	m.updateFilterSettings(map[string]string{}, "init")
+
+	// prepare configmap event handler
+	configmapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.addConfigmap,
+		UpdateFunc: m.updateConfigmap,
+		DeleteFunc: m.deleteConfigmap,
+	})
+	return m
+}
+
+// IsReady is used for checking that configuration of Yurthub has been loaded completed or not.
+// Yurthub should reject all requests from components on the node before it is ready.
+func (m *Manager) IsReady() bool {
+	return m.configMapSynced()
+}
+
+// ListAllCacheAgents is used for listing all cache agents.
+func (m *Manager) ListAllCacheAgents() []string {
+	m.RLock()
+	defer m.RUnlock()
+
+	return m.allCacheAgents.UnsortedList()
+}
+
+// IsCacheable is used for checking that http response of specified component
+// should be cached on the local disk or not.
+func (m *Manager) IsCacheable(comp string) bool {
+	if strings.Contains(comp, "/") {
+		index := strings.Index(comp, "/")
+		if index != -1 {
+			comp = comp[:index]
+		}
+	}
+
+	m.RLock()
+	defer m.RUnlock()
+	return m.allCacheAgents.HasAny("*", comp)
+}
+
+// FindFiltersFor is used for finding all filters for the specified request.
+// the return value represents all the filter names for the request.
+func (m *Manager) FindFiltersFor(req *http.Request) []string {
+	key := getKeyByRequest(req)
+	if len(key) == 0 {
+		return []string{}
+	}
+
+	m.RLock()
+	defer m.RUnlock()
+	if filters, ok := m.reqKeyToFilters[key]; ok {
+		return filters
+	}
+	return []string{}
+}
+
+func (m *Manager) addConfigmap(obj interface{}) {
+	cfg, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return
+	}
+
+	m.updateCacheAgents(cfg.Data[cacheUserAgentsKey], "add")
+	m.updateFilterSettings(cfg.Data, "add")
+}
+
+func (m *Manager) updateConfigmap(oldObj, newObj interface{}) {
+	oldCfg, ok := oldObj.(*corev1.ConfigMap)
+	if !ok {
+		return
+	}
+
+	newCfg, ok := newObj.(*corev1.ConfigMap)
+	if !ok {
+		return
+	}
+
+	if oldCfg.Data[cacheUserAgentsKey] != newCfg.Data[cacheUserAgentsKey] {
+		m.updateCacheAgents(newCfg.Data[cacheUserAgentsKey], "update")
+	}
+
+	if filterSettingsChanged(oldCfg.Data, newCfg.Data) {
+		m.updateFilterSettings(newCfg.Data, "update")
+	}
+}
+
+func (m *Manager) deleteConfigmap(obj interface{}) {
+	_, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return
+	}
+	m.updateCacheAgents("", "delete")
+	m.updateFilterSettings(map[string]string{}, "delete")
+}
+
+// updateCacheAgents update cache agents
+// todo: cache on the local disk should be removed when agent is deleted.
+func (m *Manager) updateCacheAgents(cacheAgents, action string) {
+	newAgents := make([]string, 0)
+	newAgents = append(newAgents, m.baseAgents...)
+	for _, agent := range strings.Split(cacheAgents, sepForAgent) {
+		agent = strings.TrimSpace(agent)
+		if len(agent) != 0 {
+			newAgents = append(newAgents, agent)
+		}
+	}
+
+	klog.Infof("%s cache agents: %v", action, newAgents)
```

Comment:
When action is delete, this looks like newAgents are deleted. So I think this log should be optimzed.
```suggestion
	klog.Infof("After action %s, the cache agents are as follows: %v", action, newAgents)
```

Reply:
fixed